File: 004/004.py
def ispal(x):
    x = str(x)
    for i in range(len(x)):
        if not x[i-1] == x[-i]:
            return False
    return True

greatest = 0;
for i in range(999, 100, -1):
    for j in range(999, i, -1):
        product = i*j
        if product > greatest:
            if ispal(product):
                greatest = product
        else:
            break

print(greatest)

File: 004/004.rs
fn main(){
    let mut greatest = 0;
    // let mut product = 0;
    for i in (100..999).rev(){
        for j in (i..999).rev(){
            let product = i * j;
            if product > greatest{
                if ispal(product){
                    greatest = product;
                }
            }else{
                break;
            }


        }
    }
    println!("Greatest: {}", greatest)

}

fn ispal(x: i32) -> bool{
    let word = x.to_string();
    let binding = word.as_bytes();
    let total = word.len();
    
    for idx in 0..total/2{
        if binding[idx] != binding[total-idx-1]{
            return false
        }
        
    }
    return true
}